contracts/ERC4626/VenusERC4626.sol

```diff
@@ -0,0 +1,260 @@
+
+    /// @notice Thrown when a Venus protocol call returns an error.
+    /// @dev This error is triggered if a Venus operation (such as minting or redeeming vTokens) fails.
+    /// @param errorCode The error code returned by the Venus protocol.
+    error VenusERC4626__VenusError(uint256 errorCode);
+
+    /// @notice Thrown when a deposit exceeds the maximum allowed limit.
+    /// @dev This error is triggered if the deposit amount is greater than `maxDeposit(receiver)`.
+    error ERC4626__DepositMoreThanMax();
+
+    /// @notice Thrown when a mint operation exceeds the maximum allowed limit.
+    /// @dev This error is triggered if the mint amount is greater than `maxMint(receiver)`.
+    error ERC4626__MintMoreThanMax();
+
+    /// @notice Thrown when a withdrawal exceeds the maximum available assets.
+    /// @dev This error is triggered if the withdrawal amount is greater than `maxWithdraw(owner)`.
+    error ERC4626__WithdrawMoreThanMax();
+
+    /// @notice Thrown when a redemption exceeds the maximum redeemable shares.
+    /// @dev This error is triggered if the redemption amount is greater than `maxRedeem(owner)`.
+    error ERC4626__RedeemMoreThanMax();
+
+    /// @notice Thrown when a vault action is paused.
+    /// @param action The paused vault action that triggered the error.
+    error VaultActionPaused(VaultAction action);
+
+    /// @notice Constructor to initialize the Venus ERC4626 vault.
+    /// @param asset_ The underlying ERC20 asset.
+    /// @param xvs_ The XVS token contract.
+    /// @param vToken_ The Venus vToken contract.
+    /// @param rewardRecipient_ The address receiving the XVS rewards.
+    /// @param comptroller_ The Venus Comptroller contract.
+    /// @param rewardDistributor_ The reward distributor contract.
+    constructor(
+        ERC20 asset_,
+        ERC20 xvs_,
+        VToken vToken_,
+        address rewardRecipient_,
+        IComptroller comptroller_,
+        RewardDistributorInterface rewardDistributor_
+    ) ERC4626(asset_) ERC20(_generateVaultName(asset_), _generateVaultSymbol(asset_)) {
+        VTOKEN = vToken_;
+        XVS = xvs_;
+        COMPTROLLER = comptroller_;
+        REWARD_DISTRIBUTOR = rewardDistributor_;
+        REWARD_RECIPIENT = rewardRecipient_;
+    }
+
+    /// @notice Claims XVS rewards from the reward distributor and transfers them to the reward recipient.
+    /// @dev Calls the `claimRewardToken` function on the reward distributor contract to claim rewards.
+    ///      If there is any XVS balance after claiming, it is transferred to the predefined reward recipient.
+    ///      Emits a `ClaimRewards` event with the amount of XVS claimed.
+    function claimRewards() external {
+        REWARD_DISTRIBUTOR.claimRewardToken(address(this));
+        uint256 xvsBalance = XVS.balanceOf(address(this));
+
+        if (xvsBalance > 0) {
+            XVS.safeTransfer(REWARD_RECIPIENT, xvsBalance);
+        }
+        emit ClaimRewards(xvsBalance);
```

Comment:
I think we should send them to the ProtocolShareReserve, with a new Income type: `ERC4626_WRAPPER_REWARDS`. I wouldn't hardcode the PSR address, ofc, we can keep the `REWARD_RECIPIENT` attribute, but we could "try-catch" call `REWARD_RECIPIENT.updateAssetsState` at the end of this function

Reply:
Okay, I am checking for the rewardRecipient contract size if it exist then I am calling the `REWARD_RECIPIENT.updateAssetsState`
[3306ff2](https://github.com/VenusProtocol/isolated-pools/pull/497/commits/3306ff24862b3f0f8cb2a3544d6054571336d16e)